### PR TITLE
In depth

### PR DESCRIPTION
**Pull request by ddgenome**
@jrday I'd like to merge this in to master and get this published.

### CODE REVIEW

**Comment by ddgenome** [comment]
I'd also like to rebase to clean up the history on this a bit before merging.

File: docs/user-guide/rug/generators.md [hunk 1]
```diff
@@ -1,104 +1,181 @@
...
+specifying filter words to the command. For instance, to search for generators
+for Java and Spring:
+
+```
+@atomist java spring
+```
 
-To create a project from this Rug generator, you could run the following
-[rug][cli] command:
+### Using the CLI
```
Comment (ddgenome) [inline docs/user-guide/rug/generators.md]: @Lawouach the characters between the hash marks and the capital "U" is ASCII 160, a "non-breaking space". What editor are you using that is inserting these characters? Any way to turn it off?

File: docs/user-guide/rug/generators.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+
+This will return a short list of generators, you can refine your search by
+specifying filter words to the command. For instance, to search for generators
+for Java and Spring:
+
+```
+@atomist java spring
+```
+
+### Using the CLI
```
Comment (ddgenome) [inline docs/user-guide/rug/generators.md]: @Lawouach the character between the hash marks and the capital "U" is ASCII 160, a "non-breaking space". What editor are you using that is inserting these characters? Any way to turn it off?

Comment (Lawouach) [inline docs/user-guide/rug/generators.md]: It's more likely the case that I hit the compose key followed by space. My editor should warn me about them indeed.

**Comment by ddgenome** [comment]
I've cleaned up the history and worked in @jrday's suggestions.

@Lawouach I moved your commit with the editor content to another branch, rebased on the head of this. Can you work on the branch to reorganize the text of the editor user guide to match organization of the generator user guide?

**Pull request merged**